### PR TITLE
[WIP] feat(notebook): add close and halt to notebook

### PR DESCRIPTION
**Pull request by alexandercbooth**
Adds close and halt to the notebook like in the traditional Jupyter notebook. Addresses #742.

Currently the keyboard shortcut added is `Ctrl-K`. Is this OK or would something else be more appropriate? Or perhaps using the command palette instead?

Thanks!

### CODE REVIEW

**Comment by jasongrout** [comment]
ctrl-k is used to delete to the end of a line in the editor (at least on os x), which is muscle memory for me coming from emacs. Is it Ctrl-K in the classic notebook?

**Comment by alexandercbooth** [comment]
Hi @jasongrout,

I don't believe it is. I chose it mainly for the "killing" of the kernel, not for any Jupyter specific reason. I am happy to change it, though. Is there a different shortcut you'd rather see?

Thanks!

**Comment by blink1073** [comment]
It looks like there is no shortcut in the classic notebook for "Close and Halt".   I think this is one that should not have a default keyboard shortcut at the moment.  I also use Ctrl+K the same way as @jasongrout.

**Comment by alexandercbooth** [comment]
Hi @blink1073,

OK, removed the shortcut to keep consistent with classic Jupyter. Thanks for taking a look.

File: src/notebook/plugin.ts [hunk 1]
```diff
@@ -68,6 +68,7 @@ const cmdIds = {
   restartRunAll: 'notebook:restart-runAll',
   switchKernel: 'notebook:switch-kernel',
   clearAllOutputs: 'notebook:clear-outputs',
+  killAndExit: 'notebook:halt-and-close',
```
Comment (blink1073) [inline src/notebook/plugin.ts]: Perhaps `closeAndHalt` and `close-and-halt` and `Close and Halt` to be consistent?

Comment (alexandercbooth) [inline src/notebook/plugin.ts]: Absolutely! I originally chose killAndExit to match classic Jupyter, but I agree this makes more sense here and keeps things more consistent. Thanks @blink1073!

**Comment by blink1073** [comment]
Hi @alexandercbooth, #1302 fixes a bug we had where closing a notebook automatically shut down its session, and brings back the classic notebook behavior of leaving the session available in the "Running" tab by default.  For your PR, this means that you need to run `panel.context.changeKernel(null).then(() => { panel.dispose(); }` to ensure that the session is shut down.

**Comment by alexandercbooth** [comment]
Hi @blink1073, thanks for the heads up. Just changed it to incorporate #1302. Let me know if there is anything else I can do.

Thanks for all the help!

**Comment by blink1073** [comment]
Sweet, thanks for contributing!

**Comment by alexandercbooth** [comment]
My pleasure!

**Pull request merged**